File: SdkGenerator/Model/Example/ElasticSearch/HitsMetadata.cs
using System.Text.Json.Serialization;

namespace SdkGenerator.Model.Example.ElasticSearch;

public class HitsMetadata<TDocument>
{
    [JsonPropertyName("hits")]
    public IReadOnlyCollection<Hit<TDocument>> Hits { get; init; }

    [JsonPropertyName("max_score")] public double? MaxScore { get; init; }

    [JsonPropertyName("total")]
    public long Total { get; init; }
}

File: Templates/Java/src/main/java/sdkgenerator/elastic/HitsMetadata.java
package sdkgenerator.elastic;

import com.google.gson.annotations.SerializedName;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class HitsMetadata<TDocument> {

    @NotNull
    @SerializedName("hits")
    public List<Hit<TDocument>> hits;

    @Nullable
    @SerializedName("max_score")
    public Double maxScore;

    @NotNull
    @SerializedName("total")
    public long total;

    public @NotNull List<Hit<TDocument>> getHits() {
        return hits;
    }

    public @Nullable Double getMaxScore() {
        return maxScore;
    }

    public long getTotal() {
        return total;
    }
}